### PR TITLE
feat(oracle-adapter): implement set_staleness_threshold

### LINKED ISSUE #514: [Contract] Implement `set_staleness_threshold` Function
## 📝 Description

Implement `set_staleness_threshold` that allows the admin to update the maximum age (in seconds) a price can be before it is considered stale.

**Depends on:** Initialize Oracle Adapter Contract Storage

## ✅ Requirements

- Signature: `fn set_staleness_threshold(env: Env, threshold: u64)`
- Require auth from admin address
- Assert `threshold > 0` — zero threshold is invalid
- Write `threshold` to `StalenessThreshold` storage key
- Emit `StalenessThresholdUpdated { threshold }` event

## 🎯 Acceptance Criteria

- [ ] Only admin can update the threshold
- [ ] Zero threshold panics with clear error
- [ ] New threshold immediately affects `is_price_fresh` results
- [ ] `StalenessThresholdUpdated` event emitted

## 📁 Expected files to change

- `contracts/oracle-adapter/src/lib.rs`
- `contracts/oracle-adapter/src/storage.rs`

---
> 🌿 Branch: `feat/oracle-set-staleness-threshold` | Priority: MED | Difficulty: ☕ one-coffee

---
**Priority:** MED · **Difficulty:** easy · **Track:** Contract

### PR DESCRIPTION
**Pull request by Tobi-8**
Add set_staleness_threshold function that allows admin to update the maximum age (in seconds) a price can be before stale.

- Require auth from admin address
- Panic on zero threshold with clear error
- Write threshold to StalenessThreshold storage
- Emit StalenessThresholdUpdated event
- Add tests for success, zero threshold, freshness impact, and event

closes #514

## Summary by CodeRabbit

* **New Features**
  * Administrators can now set and update the staleness threshold used to judge price data freshness.
  * Staleness threshold updates are recorded as on-chain events for transparency.
  * Zero-value thresholds are rejected, and only authorized administrators can make changes.
* **Tests**
  * Added coverage for setting, persisting, and reading the staleness threshold.
  * Verifies invalid updates fail with the expected error, non-admin attempts are blocked, and emitted events match expectations.

### CODE REVIEW

File: contracts/oracle-adapter/src/lib.rs [hunk 1]
```diff
@@ -89,6 +89,20 @@ impl OracleContract {
...
     }
 
+    pub fn set_staleness_threshold(env: Env, threshold: u64) {
+        let admin = match storage::get_admin(&env) {
+            Some(addr) => addr,
+            None => soroban_sdk::panic_with_error!(&env, OracleError::NotInitialized),
+        };
+        admin.require_auth();
+
+        assert!(threshold > 0, "threshold must be positive");
```
Comment (ryzen-xp) [inline contracts/oracle-adapter/src/lib.rs]: can  You  use OracalError  instead of  hardcode str

**Review by ryzen-xp — CHANGES_REQUESTED** [review]

**Comment by Tobi-8** [comment]
lib.rs:
- Added ThresholdZero = 3 variant to OracleError
- Replaced assert!(threshold > 0, "threshold must be positive") with if threshold == 0 { soroban_sdk::panic_with_error!(&env, OracleError::ThresholdZero); }
test.rs:
- Updated test_set_staleness_threshold_zero_fails to assert the specific OracleError::ThresholdZero error code

**Comment by Tobi-8** [comment]
Done.  Replaced assert!   with OracleError::ThresholdZero, added non-admin auth test, and rebased onto latest main to resolve conflicts. Also fixed two missing error variants (PriceNotFound, StalePrice) that were causing get_price_or_fail to fail compilation.

**Review by ryzen-xp — APPROVED** [review]
